pilot/pkg/networking/util/util.go

```diff
@@ -256,22 +256,43 @@ func ConvertLocality(locality string) *core.Locality {
 		return nil
 	}
 
-	items := strings.Split(locality, "/")
+	region, zone, subzone := splitLocality(locality)
+	return &core.Locality{
+		Region:  region,
+		Zone:    zone,
+		SubZone: subzone,
+	}
+}
+
+func LocalityToString(locality *core.Locality) string {
+	if locality == nil {
+		return ""
+	}
+
+	return locality.Region + "/" + locality.Zone + "/" + locality.SubZone
+}
+
+func LocalityMatch(proxyLocality, ruleLocality string) bool {
+	region, zone, subzone := splitLocality(proxyLocality)
+	ruleRegion, ruleZone, ruleSubzone := splitLocality(ruleLocality)
+	regionMatch := ruleRegion == "*" || region == ruleRegion
+	zoneMatch := ruleZone == "*" || ruleZone == "" || zone == ruleZone
+	subzoneMatch := ruleSubzone == "*" || ruleSubzone == "" || subzone == ruleSubzone
```

Comment:
rule can say xx, or xx/yy or xx/yy/zz. if xx only, then we set subZone to xx. if xx/yy, then we set zone=xx, subZone=yy. if xx/yy/zz, then region=xx, zone=yy, subZone = zz. Isnt that what we specified in the api?

Reply:
I meant if xx only, then set region to xx. Just as what istio currently does.  For example 
ServiceEntry Locality and pod locality (region/zone)